src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -470,46 +479,76 @@ private Long processInstallServer42Rule(ReinstallServer42FlowForSwitchManagerReq
+            return switchManager.installServer42FlowRttInputFlow(
                     dpid, command.getServer42Port(), customerPort, command.getServer42MacAddress());
+        } else if (cookie == SERVER_42_ISL_RTT_OUTPUT_COOKIE) {
+            return processInstallServer42IslRttRule(command.getSwitchId(), command.getCookie(),
+                    command.getServer42Port(), command.getServer42Vlan(), command.getServer42MacAddress());
+        } else if (new Cookie(cookie).getType() == CookieType.SERVER_42_ISL_RTT_INPUT) {
+            return processInstallServer42IslRttRule(command.getSwitchId(), command.getCookie(),
+                    command.getServer42Port(), command.getServer42Vlan(), command.getServer42MacAddress());
         } else {
             logger.warn("Skipping the installation of unexpected server 42 switch rule {} for switch {}",
                     Long.toHexString(cookie), command.getSwitchId());
             return null;
         }
     }
 
-    private void processInstallServer42Rule(InstallServer42Flow command) throws SwitchOperationException {
+    private void processInstallServer42RttRule(InstallServer42Flow command) throws SwitchOperationException {
         ISwitchManager switchManager = context.getSwitchManager();
         DatapathId dpid = DatapathId.of(command.getSwitchId().toLong());
         long cookie = command.getCookie();
 
-        if (cookie == SERVER_42_OUTPUT_VLAN_COOKIE) {
-            switchManager.installServer42OutputVlanFlow(
+        if (cookie == SERVER_42_FLOW_RTT_OUTPUT_VLAN_COOKIE) {
+            switchManager.installServer42FlowRttOutputVlanFlow(
                     dpid, command.getOutputPort(), command.getServer42Vlan(), command.getServer42MacAddress());
-        } else if (cookie == SERVER_42_OUTPUT_VXLAN_COOKIE) {
-            switchManager.installServer42OutputVxlanFlow(
+        } else if (cookie == SERVER_42_FLOW_RTT_OUTPUT_VXLAN_COOKIE) {
+            switchManager.installServer42FlowRttOutputVxlanFlow(
                     dpid, command.getOutputPort(), command.getServer42Vlan(), command.getServer42MacAddress());
-        } else if (new Cookie(cookie).getType() == CookieType.SERVER_42_INPUT) {
+        } else if (new Cookie(cookie).getType() == CookieType.SERVER_42_FLOW_RTT_INPUT) {
             PortColourCookie portColourCookie = new PortColourCookie(cookie);
             int customerPort = portColourCookie.getPortNumber();
-            switchManager.installServer42InputFlow(
+            switchManager.installServer42FlowRttInputFlow(
                     dpid, command.getInputPort(), customerPort, command.getServer42MacAddress());
+        } else if (cookie == SERVER_42_ISL_RTT_OUTPUT_COOKIE) {
+            processInstallServer42IslRttRule(command.getSwitchId(), command.getCookie(), command.getOutputPort(),
+                    command.getServer42Vlan(), command.getServer42MacAddress());
+        } else if (new Cookie(cookie).getType() == CookieType.SERVER_42_ISL_RTT_INPUT) {
```

Comment:
I'm not actually understand what is the difference between bodies of these two `if` statements. But ok, it doesn't matter.

Reply:
Merge issue. Fixed.